File: motsa_4.js
// This code beats 95.28% of users with JavaScript

/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number}
 */
var findMedianSortedArrays = function (nums1, nums2) {
  const num3 = nums1.concat(nums2).sort((a, b) => a - b);
  const l = num3.length;

  if (num3.length % 2 != 0) {
    const median = num3[(l - 1) / 2];
    return median;
  }

  const median = (num3[l / 2 - 1] + num3[l / 2]) / 2;
  return median;
};

const result = findMedianSortedArrays([1, 3], [2]);

File: rust/motsa_4.rs
// this code run 3ms Beats 55.35% of users with Rust

impl Solution {
    pub fn find_median_sorted_arrays(nums1: Vec<i32>, nums2: Vec<i32>) -> f64 {
        let mut nums3: Vec<f64> = [&nums1[..], &nums2[..]]
            .concat()
            .iter()
            .map(|&x| x as f64)
            .collect();

        nums3.sort_by(|a, b| a.partial_cmp(b).unwrap());

        let l = nums3.len();

        if l % 2 != 0 {
            nums3[(l - 1) / 2]
        } else {
            (nums3[l / 2 - 1] + nums3[l / 2]) / 2.0
        }
    }
}
